### DISCUSSION TITLE
error in pre rank: ValueError: Length mismatch: Expected axis has 3 elements, new values have 2 elements

### DISCUSSION THREAD

**Discussion by jimena2** [zqfang/GSEApy#250] [Q&A]

**Comment by zqfang** [comment]
can you print the first 5 row of your ranking ? the error message said the input ranking has 3 column

**Reply by zqfang** [reply]
delete the symbol column, and it will work

**Reply by inenj** [reply]
I'm trying the same tutorial. I deleted the symbol column as you suggested and it's giving this error:

[ERROR] No gene sets passed through filtering condition !!! 
Hint 1: Try to lower min_size or increase max_size !
Hint 2: Check gene symbols are identifiable to your gmt input.
Hint 3: Gene symbols curated in Enrichr web services are all upcases.

Can you give me an idea of what to do?

**Reply by zqfang** [reply]
the input is 2 column data frame

**Reply by inenj** [reply]
<img width="633" alt="Screenshot 2024-12-15 at 10 14 11 PM" src="https://github.com/user-attachments/assets/e29c7aea-1152-4550-8ffa-c5e0d47dc08f" />

That's what I have, so I'm not sure why it isn't working. Apologies, I'm new at this.

**Reply by zqfang** [reply]
No worries. all you need to do is discard the index
```
ranking = ranking.reset_index(drop=True).
```

then run the prerank again.

I'll add some optimization code for the case you have here in the next release

**Reply by inenj** [reply]
It worked perfectly. Thanks so much for your help!!
